### PR TITLE
Fixed NaN propagation from channel paddings on GPU

### PR DESCRIPTION
**Pull request by mwtarnowski**
Non-pointwise (i.e. kernel size != 1x1) convolutions with "SAME" padding return incorrect results when applied after element-wise division. This occurs only when both tensor operands of division have the same number of channels `C`, where `C` is not a multiply of 4, because of zero-padding applied to such tensors.

The source of the problem is performing division on channel pads which are filled by zeros - since both sides of division are zeros this operation returns `NaN` (see: The OpenCL Specification: 6.3 Operators & IEEE Standard for Floating-Point Arithmetic: 7.2 Invalid operation).

The problem is not limited to division. In case of `pow` function it is even more dangerous: instruction `pow(0., 0.)` returns `1` (see: The OpenCL Specification: 7.5 Edge Case Behavior), so there's no such a clear indicator of a problem.

### CODE REVIEW

**Comment by mwtarnowski** [comment]
To reproduce the mentioned behavior, convert the simple toy model constructed by the TensorFlow code below:
```python
import tensorflow as tf
x = tf.placeholder(tf.float32, [1,4,4,1], name='x')
y = tf.placeholder(tf.float32, [1,4,4,1], name='y')
z = tf.truediv(x, y, name='z')
t = tf.nn.conv2d(z, tf.ones([3,3,1,1]), [1,1,1,1], 'SAME', name='t')
```
with `t` as an output node and feed it with the following input values:
```python
x = np.ones([1,4,4,1])
y = np.ones([1,4,4,1])
```
Instead of the expected result, you will get the tensor of a shape `[1,4,4,1]` filled by `NaN`s.

**Comment by herbakamil** [comment]
@liyinhgqw do you know more or less when this PR will be processed - a week, two, longer?

**Comment by nolanliou** [comment]
Thanks for the great work.
